packages/rich-text/src/component/index.native.js

```diff
@@ -685,7 +690,14 @@ export class RichText extends Component {
 	}
 
 	componentWillUnmount() {
-		if ( this._editor.isFocused() && this.props.shouldBlurOnUnmount ) {
+		// `isNewRichTextAdded` flag is used to distinguish whether RichText was unmounted
+		// because of adding new RichText via pressing `enter`. If `false` it indicates
+		// block is removed and RichText should be blurred. Otherwise `blur()` is omitted
+		// for newly created RichText to avoid keyboard jumpiness.
+		if ( this.isNewRichTextAdded ) {
+			return false;
```

Comment:
```suggestion
			return;
```

We don't need to return boolean in `componentWillUnmount`